src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java

```diff
@@ -205,10 +215,14 @@ private boolean callbackResponseHasCaseAndDataClassification(CallbackResponse ca
         return defaultSecurityClassifications;
     }
 
-    private void validateAndSetData(final CaseType caseType,
+    private void validateAndSetData(final String eventId,
+                                    final CaseType caseType,
                                     final CaseDetails caseDetails,
                                     final Map<String, JsonNode> responseData) {
         caseTypeService.validateData(responseData, caseType);
+        accessControlService.verifyCreateAccess(eventId, caseType,
+            caseAccessService.getCaseCreationRoles(),
+            MAPPER.convertValue(responseData, JsonNode.class));
```

Comment:
1) AuthorisedCreateCaseOperation also does a `verifyReadAccess` other than a `verifyCreateAccess`. Shouldn't we add that here too?
2)The issue with this solution is that these callbacks methods are called not only on case creation. But also on case update. So we are changing the behaviour of case updates, which is probably something we shouldn't do.
3)We have to be careful here. When I raised this ticket I was kinda sure that we are already verify CRUD on callbacks. But it looks like that's not the case. So now we are adding CRUD validation. The problem with this is that this is a breaking change that might introduce some errors in Production